frameworks/geniverse/lib/burst-core.js

```diff
@@ -49,12 +49,19 @@
   };
 
   Burst.prototype.play = function(){
-    var deepref = this;
-    // FIXME: Is this correct behavior?
-    window.clearInterval(this.interval);
-    this.interval = window.setInterval(function(){
-      deepref.frame();
-    }, 1000 / this.fps );
+    var _this = this;
+
+    // cf. http://creativejs.com/resources/requestanimationframe/
+    function draw() {
+      _this.timeoutID = setTimeout(function() {
+        _this.requestID = requestAnimationFrame(draw);
+        // Drawing code goes here
+        _this.frame();
+      }, 1000 / this.fps);
```

Comment:
This is not actually using `fps`, because `this` no longer points to `burst-core`. If you want to use `fps`, you have to change it to `_this.fps`.

However, doing this massively slows down the animation, because 10 fps is really slow.

Instead, the way it was working was to essentially ignore the `setTimeout` (since it was being passed a value of `NaN`, which leads it to trigger instantly), and I think that's the correct way to do it in this case. There's no need to try and limit the framerate in this animation -- even on our fast machines, this animation is so slow that it's not worth slowing down further.

I think the best thing to do would be to scrap the `setTimeout` here and just use `requestAnimationFrame`.

```
    function draw() {
      _this.requestID = requestAnimationFrame(draw);
      _this.frame();
    }
    // don't start a timer if we've already got one
    if (!this.requestID)
      draw();
```